server/src/main/java/org/opensearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -754,6 +754,9 @@ public void sendSearchResponse(InternalSearchResponse internalSearchResponse, At
             }
             searchRequestContext.setTotalHits(internalSearchResponse.hits().getTotalHits());
             searchRequestContext.setShardStats(results.getNumShards(), successfulOps.get(), skippedOps.get(), failures.length);
+            searchRequestContext.successfulSearchShardIndices(
+                results.getSuccessfulResults().map(result -> result.getSearchShardTarget().getIndex()).collect(Collectors.toList())
+            );
```

Comment:
We should collect this in `Set` instead of `List` to avoid duplicates

Reply:
Good point